src/pc/sotn.c

```diff
@@ -204,12 +206,9 @@ bool InitGame(void) {
         return false;
     }
 
-    // TODO different between RIC and ARC
-    if (!FileAsString(
-            InitBlueprintData, "assets/dra/factory_blueprint.json", NULL)) {
-        ERRORF("failed to init blueprint data");
-        return false;
-    }
+    InitBlueprintData("assets/dra/factory_blueprint.json", g_FactoryBlueprints);
```

Comment:
Why not keep using `FileAsString` and pass either `g_RicFactoryBlueprints` or `g_FactoryBlueprints` as `param` instead of NULL` as a destination? Then instead of doing `FactoryBlueprint* item = &g_FactoryBlueprints[i];` you can do

```c
FactoryBlueprint* array = (FactoryBlueprint*)file->param;
...
FactoryBlueprint* item = array[i];
```

Reply:
Ok seems to work